### PR TITLE
V2 roggc - QRScan Component

### PR DESCRIPTION
**Pull request by roggc**
make a QRScan component in storybook

### CODE REVIEW

**Comment by roggc** [comment]
qrscan

**Comment by ralph-dev** [comment]
Heads up the footer beside the phone is missing the text and other phones

File: src/Containers/QRScan.tsx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    ResponsiveInterface,
+} from '../Utils/BaseStyles';
+
+export interface QRScanProps
+    extends MainInterface,
+        ResponsiveInterface,
+        React.HTMLAttributes<HTMLDivElement> {
+    Icon: React.FC;
+    title: string;
+    InFooter: React.FC;
```
Comment (ralph-dev) [inline src/Containers/QRScan.tsx:17]: Convert this into text and icon separately so you can embedded this into controls.

File: src/Containers/QRScan.tsx [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    flex: 1;
+`;
+
+const QRIconContainer = styled(MiddleContainers)`
+    // border: 1px solid brown;
+    margin-right: 2px;
+`;
+
+const ContentRightContainer = styled(MiddleContainers)`
+    // border: 1px solid blue;
```
Comment (ralph-dev) [inline src/Containers/QRScan.tsx:66]: Remove commented code

File: src/Containers/QRScan.tsx [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+        border-radius: 0 0 15px 15px;
+    }
+`;
+
+export interface SnackpassProps
+    extends MainInterface,
+        ResponsiveInterface,
+        React.HTMLAttributes<HTMLDivElement> {}
+
+export const Snackpass: React.FC<SnackpassProps> = ({}): React.ReactElement => (
```
Comment (ralph-dev) [inline src/Containers/QRScan.tsx:163]: Name this something more generic like QR Promo Card

File: src/Containers/QRScan.tsx [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+`}
+`;
+
+const TableNumber = styled.div`
+    ${({ theme }): string => `
+font-size:${theme.font.size.h1};
+`}
+`;
+
+export const QRURL1 =
```
Comment (ralph-dev) [inline src/Containers/QRScan.tsx:199]: Constants should be at the top

**Review by ralph-dev — CHANGES_REQUESTED** [review]
Needs work

**Comment by fpunny** [comment]
just random question, but does it not seem kinda sketch that it has snackpass on it?

**Comment by ralph-dev** [comment]
> just random question, but does it not seem kinda sketch that it has snackpass on it?

It's just a practice exercise to get him familiar with storybook, but yeah we should remove the text before merging so the user can pass in any text / change it to CheaprEats. We will do this at the end after approval to make sure concepts are learned first

File: stories/Containers/QRScan.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { QRScan, QRScanProps, PhoneBox, qrImgURL1 } from '../../src';
+import { createStoryTitle } from '../Constants';
+import { Meta, Story } from '@storybook/react';
+
+import { MainTheme } from '../../src/Themes';
+
+export default {
+    title: createStoryTitle('QRScan'),
+    component: QRScan,
```
Comment (ralph-dev) [inline stories/Containers/QRScan.stories.tsx:10]: Put the args here for duplicate args btw that way you don't need to redeclare everytime

**Review by ralph-dev — APPROVED** [review]
LGTM

File: stories/Containers/QRScan.stories.tsx (hunk 5 above)
Comment (roggc) [reply stories/Containers/QRScan.stories.tsx:10]: ok, will try and commit again...

Comment (roggc) [reply stories/Containers/QRScan.stories.tsx:10]: I tried but it doesn't work as expected. they don't work as default args. if I don't put them outside the default they don't appear on the component.

Comment (ralph-dev) [reply stories/Containers/QRScan.stories.tsx:10]: They should work, have you looked at the other components that have implemented this (Look at Button)

Comment (roggc) [reply stories/Containers/QRScan.stories.tsx:10]: Ok, I made a separate object defaultArgs and used js destructuring. committed it.

**Comment by ralph-dev** [comment]
Feel free to merge once this PR is ready (If you received an LGTM you are all clear)

**Pull request merged**